Sphere/Sphere.Godot/Nodes/MainNode.cs

```diff
@@ -0,0 +1,150 @@
+
+            var serviceCollection = new ServiceCollection();
+
+            var serviceProvider = serviceCollection
+                .RegisterCommon()
+                .RegisterServices()
+                .BuildServiceProvider();
+
+            Console.WriteLine("End service registration...");
+
+            var logger = serviceProvider.GetRequiredService<ILogger<MainNode>>();
+
+            var server = serviceProvider.GetRequiredService<IServer>();
+            this.AddChild((Node)server);
+
+            server.StartAsync();
+            
+            Task.Run(() =>
+            {
+                while (running)
+                {
+                    var command = Console.ReadLine();
+
+                    switch (command)
+                    {
+                        case "stop":
+                            server.StopAsync();
+                            break;
+                        case "start":
+                            server.StartAsync();
+                            break;
+                        case "exit":
+                            server.StopAsync();
+                            running = false;
+                            break;
+                    }
+                }
+
+                ((Node)server).QueueFree();
+                this.QueueFree();
+            });
+        }
+    }
+
+    public partial class Server : Node, IServer
+	{
+        private readonly TcpServer _listener;
+        private readonly ILogger<Server> _logger;
+        private readonly IOptions<ServerConfiguration> _serverConfiguration;
+        private PackedScene _clientScene;
+        private Dictionary<int, IClient> _portClientMap = new Dictionary<int, IClient>();
+
+        public Server(ILogger<Server> logger, IOptions<ServerConfiguration> options)
+		{
+			_listener = new TcpServer();
+            _logger = logger;
+            _serverConfiguration = options;
+
+            _clientScene = (PackedScene)ResourceLoader.Load("res://Client.tscn");
+        }
+
+        public override async void _Ready()
+        {
+            _listener.Listen(_serverConfiguration.Value.Port);
+            _logger.LogInformation("Started server on port: [{port}]", _serverConfiguration.Value.Port);
+        }
+
+        public override async void _Process(double delta)
+        {
+            while (_listener.IsListening())
+            {
+                var tcpClient = _listener.TakeConnection();
+
+                if (tcpClient != null)
+                {
+                    _logger.LogInformation("Connection accepted on port: [{port}]", tcpClient.GetLocalPort());
+
+                    await HandleConnection(tcpClient);
+                }
+            }
+        }
+
+        private async Task HandleConnection(StreamPeerTcp client)
+        {
+            _logger.LogInformation("Handle connection started");
+
+            client.SetNoDelay(true);
+
+            var clientNode = _clientScene.Instantiate<ClientNode>();
```

Comment:
You need to `AddChild(clientNode)`, either as top-level node (to root) or to the server itself. Otherwise it will be dangling and will not get any updates (so no _Process calls, etc).

You can also call `SetupClient` here

Reply:
sure, I didn't get to proper client handling so far